sdk/identity/identity/src/msal/browserFlows/msalBrowserCommon.ts

```diff
@@ -74,152 +47,322 @@ export function defaultBrowserMsalConfig(
-   * Attempts to retrieve the token in the browser.
+   * Attempts to retrieve the token in the browser through interactive methods.
    */
-  protected abstract doGetToken(scopes: string[]): Promise<AccessToken>;
+  async function getTokenInteractive(
+    scopes: string[],
+    getTokenOptions?: CredentialFlowGetTokenOptions,
+  ): Promise<AccessToken> {
+    const activeAccount = await getActiveAccount();
+    if (!activeAccount) {
+      throw new AuthenticationRequiredError({
+        scopes,
+        getTokenOptions,
+        message:
+          "Silent authentication failed. We couldn't retrieve an active account from the cache.",
+      });
+    }
+
+    const parameters: msalBrowser.RedirectRequest = {
+      authority: getTokenOptions?.authority || msalConfig.auth.authority!,
+      correlationId: getTokenOptions?.correlationId,
+      claims: getTokenOptions?.claims,
+      account: publicToMsal(activeAccount),
+      loginHint: loginHint,
+      scopes,
+    };
+    const msalApp = await getApp();
+    switch (loginStyle) {
+      case "redirect":
+        // This will go out of the page.
+        // Once the InteractiveBrowserCredential is initialized again,
+        // we'll load the MSAL account in the constructor.
+
+        await msalApp.acquireTokenRedirect(parameters);
+        return { token: "", expiresOnTimestamp: 0, tokenType: "Bearer" };
+      case "popup":
+        return handleResult(scopes, await app.acquireTokenPopup(parameters));
+    }
+  }
 
   /**
-   * Attempts to retrieve an authenticated token from MSAL.
+   * Attempts to get token through the silent flow.
+   * If failed, get token through interactive method with `doGetToken` method.
    */
-  public async getToken(
+  async function getToken(
     scopes: string[],
-    options: CredentialFlowGetTokenOptions = {},
+    getTokenOptions: CredentialFlowGetTokenOptions = {},
   ): Promise<AccessToken> {
-    const tenantId =
-      processMultiTenantRequest(this.tenantId, options, this.additionallyAllowedTenantIds) ||
-      this.tenantId;
+    const getTokenTenantId =
+      processMultiTenantRequest(tenantId, getTokenOptions, additionallyAllowedTenantIds) ||
+      tenantId;
 
-    if (!options.authority) {
-      options.authority = getAuthority(tenantId, this.authorityHost);
+    if (!getTokenOptions.authority) {
+      getTokenOptions.authority = getAuthority(getTokenTenantId, authorityHost);
     }
 
     // We ensure that redirection is handled at this point.
-    await this.handleRedirect();
+    await handleRedirect();
 
-    if (!(await this.getActiveAccount()) && !this.disableAutomaticAuthentication) {
-      await this.login(scopes);
+    if (!(await getActiveAccount()) && !disableAutomaticAuthentication) {
+      await login(scopes);
     }
-    return this.getTokenSilent(scopes).catch((err) => {
+
+    // Attempts to get the token silently; else, falls back to interactive method.
+    try {
+      return getTokenSilent(scopes, getTokenOptions);
```

Comment:
You need to `await` here in order to catch and handler the error 👍

Reply:
Ah good catch, thank you!